lib/puma/const.rb

```diff
@@ -1,6 +1,8 @@
 #encoding: utf-8
 # frozen_string_literal: true
 
+require "set"
```

Comment:
Do we need to use `Set`? We should be careful what dependencies we introduce.

[Ruby 3.0](https://www.ruby-lang.org/en/news/2020/12/25/ruby-3-0-0-released/) made `set` a default gem (and there was also a recent change, can't remember/find exactly when, where you no longer need to require set). Anyway, I can't say for sure if this will be a problem, but similar cases [have been](https://github.com/puma/puma/discussions/3034).

Reply:
As far as I can see, `Set` has been available since [at least Ruby 2.3](https://docs.ruby-lang.org/en/2.3.0/Set.html), so the Ruby versions supported by puma 7 will have it.

It seems to me that a set is the appropriate data structure here. We could use a `Hash`, but it will have an uglier interface, or we could use an `Array`, but it will have worse performance (about 4-5 times slower). However, that performance difference might be negligible compared to the rest of the code.

I'll let you decide, just tell me! :slightly_smiling_face:

Reply:
What about making the option a proc? That way the implementation is not our responsibility, and the users are free to use any data structure they want. We can still recommend using a set in the example, since most users will have that available to them.

Thoughts? :upside_down_face: